app/serializers/-shoebox.js

```diff
@@ -1,6 +1,17 @@
 import DS from 'ember-data';
 
 export default DS.JSONAPISerializer.extend({
+  // We override _normalizeResourceHelper so that resources in the shoebox are normalized
+  // using the same serializer that serialized them (i.e., the shoebox serializer). Otherwise,
+  // this method would try to normalize them by deferring to model-specific serializers.
+  _normalizeResourceHelper(resourceHash) {
+    let modelName = this.modelNameFromPayloadKey(resourceHash.type);
+    let modelClass = this.store.modelFor(modelName);
```

Comment:
Literally copy/pasted from ember-data. The only changed line is the one that sets the serializer to `this`. (I also removed some warnings and assertions.)